synapse_spamcheck_badlist/bad_list_filter.py

```diff
@@ -21,6 +21,7 @@
 from twisted.internet import defer, reactor
 from twisted.internet.threads import deferToThread
 
+from synapse.logging.context import make_deferred_yieldable
```

Comment:
This can be imported from Synapse's ModuleApi via:

```python
from synapse.module_api import make_deferred_yieldable
```

The full list of proxied objects can be seen here: https://github.com/matrix-org/synapse/blob/78a70a2e0bdff5bdf02ba093eefab0b16ebefd73/synapse/module_api/__init__.py#L63-L77